synapse/lib/types.py

```diff
@@ -624,11 +624,20 @@ class Hex(Type):
 
     _opt_defs = (
         ('size', 0),  # type: ignore
-        ('zeropad', False),
+        ('zeropad', 0),
     )
 
     def postTypeInit(self):
         self._size = self.opts.get('size')
+
+        # This is for backward compat with v2.142.x where zeropad was a bool
+        self._zeropad = self.opts.get('zeropad')
+        if isinstance(self._zeropad, bool):
```

Comment:
While i dig the fancy syntax, this would probably be clearer as an if/else. Type construction isn't perf sensitive :)